src/plugins/event-plugins/NavigationPlugin.ts

```diff
@@ -8,288 +11,100 @@ import {
-                    : 0,
-            navigationTimingLevel: 2
-        };
-
-        if (this.context?.record) {
-            this.context.record(
-                PERFORMANCE_NAVIGATION_EVENT_TYPE,
-                eventDataNavigationTimingLevel2
-            );
-        }
+            // Record
+            const e = entry as PerformanceNavigationTiming;
+            this.context?.record(PERFORMANCE_NAVIGATION_EVENT_TYPE, {
+                version: '2.0.0',
+                name: this.context.config.recordResourceUrl
+                    ? e.name
+                    : undefined,
+                entryType: NAVIGATION,
+                startTime: e.startTime,
+                duration: e.duration,
+                initiatorType: e.initiatorType,
+                nextHopProtocol: e.nextHopProtocol,
+                workerStart: e.workerStart,
+                redirectStart: e.redirectStart,
+                redirectEnd: e.redirectEnd,
+                fetchStart: e.fetchStart,
+                domainLookupStart: e.domainLookupStart,
+                domainLookupEnd: e.domainLookupEnd,
+                connectStart: e.connectStart,
+                connectEnd: e.connectEnd,
+                secureConnectionStart: e.secureConnectionStart,
+                requestStart: e.requestStart,
+                responseStart: e.responseStart,
+                responseEnd: e.responseEnd,
+                transferSize: e.transferSize,
+                encodedBodySize: e.encodedBodySize,
+                decodedBodySize: e.decodedBodySize,
+                serverTiming: e.serverTiming
+                    ? e.serverTiming.map(
+                          (s) => s as PerformanceServerTimingPolyfill
+                      )
+                    : undefined,
+                domComplete: e.domComplete,
+                domContentLoadedEventEnd: e.domContentLoadedEventEnd,
+                domContentLoadedEventStart: e.domContentLoadedEventStart,
+                domInteractive: e.domInteractive,
+                loadEventEnd: e.loadEventEnd,
+                loadEventStart: e.loadEventStart,
+                redirectCount: e.redirectCount,
+                type: e.type,
+                unloadEventEnd: e.unloadEventEnd,
+                unloadEventStart: e.unloadEventStart
+            } as NavigationEvent);
+
+            // Teardown
+            this.po.disconnect();
+        });
     };
 
-    /**
-     * loadEventEnd is populated as 0 if the web page has not loaded completely, even though LOAD has been fired.
-     * As a result, if loadEventEnd is populated, we can ignore eventListener and record the data directly.
-     * On the other hand, if not, we have to use eventListener to initializes PerformanceObserver.
-     * PerformanceObserver will act as a second check for the final load timings.
-     */
     protected onload(): void {
-        if (this.enabled) {
-            if (this.hasTheWindowLoadEventFired()) {
```

Comment:
Checking for load event is not necessary anymore, since the `po` checks the buffer.

Reply:
However, I have left the load event tests in case there's a regression.